test/gtest/include/common-tests.hpp

```diff
@@ -322,4 +350,36 @@ inline std::ostream &operator<<(std::ostream &os,
   return os;
 }
 
+template <typename... Ts>
+inline std::ostream &operator<<(std::ostream &os,
+                                const std::tuple<Ts...> &obj) {
+  os << fmt::format("{}", obj);
+  return os;
+}
+
+template <typename T1, typename T2>
+inline std::ostream &operator<<(std::ostream &os,
+                                const std::pair<T1, T2> &obj) {
+  os << fmt::format("{}", obj);
+  return os;
+}
+
 } // namespace DR_RANGES_NAMESPACE
+
+namespace std {
+
```

Comment:
@BenBrock if the overloads are in std, do we still need the same thing in rng and shp?

Reply:
Can we avoid putting things to "std" namespace?

Reply:
gtest `EXPECT_EQ` requires that the operands can be comparable with `operator==` and printed with `operator<<`. 
To find these operators with ADL, you put it in `std`. Otherwise, you have to put the operator in multiple namespaces. This is only for the testing code so it will not pollute std for regular use of the library.